java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java

```diff
@@ -869,49 +869,105 @@ static String getPhaseTapChangerRegulatedSide(TwoWindingsTransformer transformer
     }
 
     static NetworkDataframeMapper identifiables() {
-        return NetworkDataframeMapperBuilder.ofStream(network -> network.getIdentifiables().stream(), NetworkDataframes::getT2OrThrow)
+        return NetworkDataframeMapperBuilder.ofStream(network -> network.getIdentifiables().stream(),
+                        getOrThrow(Network::getIdentifiable, "Identifiable"))
                 .stringsIndex("id", Identifiable::getId)
                 .strings("type", identifiable -> identifiable.getType().toString())
                 .build();
     }
 
     static NetworkDataframeMapper injections() {
         return NetworkDataframeMapperBuilder.ofStream(network -> network.getConnectableStream()
-                        .filter(connectable -> connectable instanceof Injection),
-                NetworkDataframes::getT2OrThrow)
+                                .filter(connectable -> connectable instanceof Injection),
+                        NetworkDataframes::getT2OrThrow)
                 .stringsIndex("id", Connectable::getId)
                 .strings("type", connectable -> connectable.getType().toString())
-                .strings("voltage_level_id", connectable -> ((Injection) connectable).getTerminal().getVoltageLevel().getId())
-                .strings("bus_id", connectable -> ((Injection) connectable).getTerminal().getBusView().getBus() == null ? "" :
-                        ((Injection) connectable).getTerminal().getBusView().getBus().getId())
+                .strings("voltage_level_id", connectable -> ((Injection<?>) connectable).getTerminal().getVoltageLevel().getId())
+                .strings("bus_id", connectable -> ((Injection<?>) connectable).getTerminal().getBusView().getBus() == null ? "" :
+                        ((Injection<?>) connectable).getTerminal().getBusView().getBus().getId())
                 .build();
     }
 
     static NetworkDataframeMapper branches() {
-        return NetworkDataframeMapperBuilder.ofStream(Network::getBranchStream, NetworkDataframes::getT2OrThrow)
+        return NetworkDataframeMapperBuilder.ofStream(Network::getBranchStream, getOrThrow(Network::getBranch, "Branch"))
```

Comment:
Not part of this PR but it would be better with an enum here too

Reply:
why do we need an enum here for ?

Reply:
Mainly so that people don't have to go and read the method implementation to know what are the possible values